release/nightly_tests/dataset/map_batches_benchmark.py

```diff
@@ -0,0 +1,128 @@
+from typing import Literal, Optional, Union
+
+import ray
+from ray.data._internal.compute import ActorPoolStrategy, ComputeStrategy
+from ray.data.dataset import Dataset
+
+from benchmark import Benchmark
+
+
+def map_batches(
+    input_ds: Dataset,
+    batch_size: int,
+    batch_format: Literal["default", "pandas", "pyarrow", "numpy"],
+    compute: Optional[Union[str, ComputeStrategy]] = None,
+    num_calls: Optional[int] = 1,
+) -> Dataset:
+
+    ds = input_ds
+    for _ in range(num_calls):
+        ds = ds.map_batches(
+            lambda ds: ds,
+            batch_format=batch_format,
+            batch_size=batch_size,
+            compute=compute,
+        )
+    return ds
+
+
+def run_map_batches_benchmark(benchmark: Benchmark):
+    input_ds = ray.data.read_parquet(
+        "s3://shuffling-data-loader-benchmarks/data/input_data_0.parquet.snappy"
```

Comment:
Are we feeling confident about this being a relatively "representative" dataset, with it being synthetic user data? Another option would be a popular open tabular dataset like the NYC taxi dataset.

Reply:
Also, this will only read a single file and transform a single block (assuming the file is smaller than the block splitting size); are we wanting to have at least some parallelism for this single-node benchmark? Otherwise, we're not really benchmarking anything meaningful in the compute layer.

(This could be a follow-up.)